prov/util/src/util_mem_monitor.c

```diff
@@ -140,9 +150,9 @@ int ofi_monitor_add_cache(struct ofi_mem_monitor *monitor,
 	pthread_mutex_lock(&monitor->lock);
 	if (dlist_empty(&monitor->list)) {
 		if (monitor == uffd_monitor)
-			ret = ofi_uffd_init();
+			ret = ofi_uffd_start();
 		else if (monitor == memhooks_monitor)
-			ret = ofi_memhooks_init();
+			ret = ofi_memhooks_start();
```

Comment:
Should start and stop also be a memory monitor op?

Reply:
I wasn't sure, but we can add them there if it ends up making sense.

Reply:
I think it makes sense. Then when changing to multiple monitors per MR cache, start/stop can be called in a for loop.

Reply:
There ends up being 2 arrays for monitors.  One array is a list of all available monitors.  The other is a list of monitors per hmem_iface -- unless, there's a reason/way to support more than 1 monitor per hmem_iface.

Right now, start is dependent on environment variable settings, which does not lend itself to being called in a loop without additional infrastructure.  And it's not clear if that extra infrastructure is worth the effort to me.  Stop could probably be called in a loop with only some minor changes to make it a no-op if start was never called.

Reply:
Looks like I have actually implemented the loop in the MR cache code instead of the monitor code.

Maybe I am missing something.... but the monitor start functions do not appear dependent on env variables. It looks like initialization and setting default memory monitor are dependent on the env variables.

Since we know the the monitor is not NULL, it seems like this block could be simplified to 
```
if (!monitor)
	return -FI_ENOSYS;

pthread_mutex_lock(&monitor->lock);
if (dlist_empty(&monitor->list)) {
	ret = monitor->start(monitor);
	if (ret)
		goto out;
}
```

Reply:
I guess the env var is already setting up the extra infrastructure.  In this case by setting monitor to the correct value.  Monitor is acting as the hmem_iface monitor array, which is this case is limited to size 1.